pkg/utils/utils_test.go

```diff
@@ -0,0 +1,54 @@
+package utils
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestChanFromSlice(t *testing.T) {
+	// requireReceive is a helper function to check if a value can immediately be received from a channel.
+	requireReceive := func(t *testing.T, ch <-chan int, expected int) {
+		t.Helper()
+
+		select {
+		case v, ok := <-ch:
+			require.True(t, ok, "receiving should return a value")
+			require.Equal(t, expected, v)
+		default:
+			require.Fail(t, "receiving should not block")
+		}
+	}
+
+	// requireReceive is a helper function to check if the channel is closed and empty.
+	requireClosedEmpty := func(t *testing.T, ch <-chan int) {
+		t.Helper()
+
+		select {
+		case _, ok := <-ch:
+			require.False(t, ok, "receiving from channel should not return anything")
+		default:
+			require.Fail(t, "receiving should not block")
+		}
+	}
+
+	t.Run("Nil", func(t *testing.T) {
+		ch := ChanFromSlice[int](nil)
+		require.NotNil(t, ch)
+		requireClosedEmpty(t, ch)
+	})
+
+	t.Run("Empty", func(t *testing.T) {
+		ch := ChanFromSlice([]int{})
```

Comment:
_In practice_ `[]T{}!=nil`. But could this change _in theory_ so we should use `make([]T,0,1)` as in TestBinary_MarshalJSON, just to be sure to cover both nil and empty?

Wait! _I think_ we should use `make([]T,0,1)` anyway as it's the absolute corner case of an empty slice.

Reply:
Admittedly (luckily!) you're right.

> Wait! _I think_ we should use `make([]T,0,1)` anyway as it's the absolute corner case of an empty slice.

But I think we should do this anyway as `cap([]int{})==0` in contrast to make(). Just to provoke the tested code a bit (which is one of the purposes of unit tests).